### ISSUE TITLE
FormDataSet not properly serializing text/plain files

### ISSUE THREAD

**Issue by johnjaylward** [restlet/restlet-framework-java#1006]
if I want to attach a file to a form post that is MediaType.TEXT_PLAIN, the serializer improperly encodes it as a form field with no file name in the disposition:

```
final Request request = new Request(Method.POST, ...);
final FormDataSet fds = new FormDataSet();
fds.setMultipart(true);
final FormData fileRep = new FormData("file", new FileRepresentation(new File("test.txt"), MediaType.TEXT_PLAIN));
fds.getEntries().add(fileRep);
request.setEntity(fds);
final Response response = router.handle(request);
```

This creates a form part with:
```
Content-Disposition: form-data; name="name";
Content-Type: text/plain; encoding=UTF-8
```

the correct form part should be:
```
Content-Disposition: form-data; name="file"; filename="test.txt"
Content-Type: text/plain; encoding=UTF-8
```

When the Apache FileUpload parser reads the post, there is no filename, and the "formField" property is set to true.

If I set the MediaType to MediaType.APPLICATION_OCTET_STREAM, it works as it should, generating the following disposition:
```
Content-Disposition: form-data; name="file"; filename="test.txt"
Content-Type: application/octet-stream
```

**Comment by thboileau** [comment]
Hi,

thanks for reporting this issue.
The ug is located in the FormDataSet class, it sets the disposition filename according to the media type of the representation which is should be fixed. I take care of that.

**Comment by thboileau** [comment]
Thanks for reporting this issue.
I've fixed the 2.3 and master (futur 3.0) branches.
